File: standard_algorithms/knapsack_0_1.py

#list of weights
w = [0,2,3,3,4,5]
#list of values
v = [0,3,4,3,5,8]

#knapsack capacity
W = 15

M = [ [ 0 for i in range(0,W+1) ] for j in range(0,len(w))]



for i in range(1,len(w)):
    for j in range(1,W+1):
        if w[i] > j:
            M[i][j] = M[i-1][j]
        else:
            M[i][j] = max(M[i-1][j], M[i-1][j-w[i]] + v[i])

print("Answer is: ")            
print(M[-1][-1])

File: julia/standard_algorithms/knapsack_0_1.jl
#list of weights
w = [2,3,3,4,5]
#list of values
v = [3,4,3,5,8]

#knapsack capacity
W = 15

M = [ [ 0 for i in 1:W+1 ] for j in 1:length(w)+1]

#=
M[1][1] = 0
...
M[1][16] = 0
=#
#problematic, while indexing starts from 1

for i in 1:length(w)
    for j in 1:W+1
        if w[i] > j-1
            M[i+1][j] = M[i][j]
        else
            M[i+1][j] = max(M[i][j], M[i][j-w[i]] + v[i])
		end
	end
end

println("Answer is: ")
print(last(last(M)))